File: app/src/main/java/com/bnotion/uniuyotowncampus/MainActivity.java
package com.bnotion.uniuyotowncampus;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
private CardView mDirections, mSearch, mNotifications, mStudySpace, mFeedback, mCampusInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // locking out landscape screen orientation for mobiles
        if(getResources().getBoolean(R.bool.portrait_only)){
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
            // locking out portait screen orientation for tablets
        } if(getResources().getBoolean(R.bool.landscape_only)) {
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
        }
        setContentView(R.layout.activity_main);
        mDirections = findViewById(R.id.directions);
        mSearch = findViewById(R.id.search);
        mNotifications = findViewById(R.id.notifications);
        mStudySpace = findViewById(R.id.study_space);
        mFeedback = findViewById(R.id.feedback);
        mCampusInfo = findViewById(R.id.campus_info);
        mDirections.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, DirectionsActivity.class);
            startActivity(intent);
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        });
        mSearch.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, SearchActivity.class);
            startActivity(intent);
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        });
        mNotifications.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, NotificationsActivity.class);
            startActivity(intent);
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        });
        mStudySpace.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, StudySpaceActivity.class);
            startActivity(intent);
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        });
        mFeedback.setOnClickListener(v -> {
            try {
                String uriText =
                        "mailto:bnotionsoftware@gmail.com" +
                                "?subject=" + Uri.encode("Feedback for app") +
                                "&body=" + Uri.encode("info");
                Uri uri = Uri.parse(uriText);
                Intent emailIntent = new Intent(Intent.ACTION_SENDTO);
                emailIntent.setData(uri);
                startActivity(Intent.createChooser(emailIntent, "Send email using..."));
            } catch (android.content.ActivityNotFoundException ex) {
                Toast.makeText(MainActivity.this, "No email clients installed.", Toast.LENGTH_SHORT).show();
            }
        });
        mCampusInfo.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this, CampusInfoActivity.class);
            startActivity(intent);
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        });
    }
}

File: app/src/main/java/com/bnotion/uniuyotowncampus/MainActivity.kt
package com.bnotion.uniuyotowncampus

import android.content.Intent
import android.content.pm.ActivityInfo
import android.net.Uri
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Lock screen orientation
        if (resources.getBoolean(R.bool.portrait_only)) {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        }
        if (resources.getBoolean(R.bool.landscape_only)) {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        }

        setContentView(R.layout.activity_main)

        val directionsCard: CardView = findViewById(R.id.directions)
        val searchCard: CardView = findViewById(R.id.search)
        val notificationsCard: CardView = findViewById(R.id.notifications)
        val studySpaceCard: CardView = findViewById(R.id.study_space)
        val feedbackCard: CardView = findViewById(R.id.feedback)
        val campusInfoCard: CardView = findViewById(R.id.campus_info)

        directionsCard.setOnClickListener {
            startActivity(Intent(this, DirectionsActivity::class.java))
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        }

        searchCard.setOnClickListener {
            startActivity(Intent(this, SearchActivity::class.java))
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        }

        notificationsCard.setOnClickListener {
            startActivity(Intent(this, NotificationsActivity::class.java))
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        }

        studySpaceCard.setOnClickListener {
            startActivity(Intent(this, StudySpaceActivity::class.java))
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        }

        feedbackCard.setOnClickListener {
            try {
                val uriText = "mailto:bnotionsoftware@gmail.com" +
                        "?subject=${Uri.encode("Feedback for app")}" +
                        "&body=${Uri.encode("info")}"
                val uri = Uri.parse(uriText)
                val emailIntent = Intent(Intent.ACTION_SENDTO).apply {
                    data = uri
                }
                startActivity(Intent.createChooser(emailIntent, "Send email using..."))
            } catch (ex: android.content.ActivityNotFoundException) {
                Toast.makeText(this, "No email clients installed.", Toast.LENGTH_SHORT).show()
            }
        }

        campusInfoCard.setOnClickListener {
            startActivity(Intent(this, CampusInfoActivity::class.java))
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
        }
    }
}
